### ISSUE TITLE
problems with logging and processing logs using Google Forms

### ISSUE THREAD

**Issue by Koloydenko** [swirldev/swirl#738]
Hi

I find swirl quite useful, and hope to give  a few sessions to undergraduate students using it. However, I am struggling to make the logging work with google forms, hoping someone may be able to help me fix the following problem: 
When I use the example https://github.com/seankross/Google_Forms_Course and call 
swirlify::google_form_decode("test.csv")
I get
> swirlify::google_form_decode("test.csv")
Error: 'W�{.�' does not exist in current working directory ('/home/utah113/R/x86_64-pc-linux-gnu-library/3.4/swirl/Courses/Google_Forms_Course/Lesson_1').
In addition: Warning messages:
1: In grepl("\n", x) : input string 1 is invalid in this locale
2: In grepl("\n", path) : input string 1 is invalid in this locale
3: In grepl("^((http|ftp)s?|sftp)://", path) :
  input string 1 is invalid in this locale
4: In grepl("^(/|[A-Za-z]:|\\\\|~)", path) :
  input string 1 is invalid in this locale
[test_csv_and_customTests_R.zip](https://github.com/swirldev/swirl/files/2705179/test_csv_and_customTests_R.zip)

I am attaching the google form csv file (which, I believe, failed to store the complete log and contains only the respondent's manually entered input and the time stamp), and the customTests.R file, where I replaced the original link by the one to my google form. The other files from the original example were used unchanged.  I have also tried another example and have changed working directories, but the same problem persists.  I have also emailed the developers at info@swirlstats.com.  Any help will be appreciated!

Alexey

**Comment by Koloydenko** [comment]
I have actually made some progress trying to make this work.  It seems that the Google Forms have been upgraded after the instructions on https://github.com/seankross/Google_Forms_Course were issued. Previously, I was getting an empty content in the form most likely because I ignored google's instruction "Pre-fill responses, then click "Get link"" that came in response to ''Get pre-filled link''. While it is still unclear to me what exactly I should pre-fill responses with, I tried entering a few characters in the 'My answer' field. The result is that this text got appended to the generated link, which resulted in the log of the lesson session going automatically into the form (previously, the form would just pop up prompting the user for input); the form would still expect the user to press 'Submit', though, but now I can extract meaningful log information from the cvs file. Now, it seems that depending on what exactly I pre-fill my responses with determines whether or not the form will work. Specifically, it has worked for me (I will show an output below) when I pre-filled the responses with just a single character, '1'. Below is a typical output which seems to be very reasonable to me:
> swirlify::google_form_decode("Untitled_form0.csv")
         user         course_name lesson_name question_number correct attempt skipped   datetime
1      MrEasy Google Forms Course    Lesson 1               2    TRUE       1   FALSE 1548184050
2      MrEasy Google Forms Course    Lesson 1               3      NA       1      NA 1548184054
3      MrEasy Google Forms Course    Lesson 1               2   FALSE       1   FALSE 1548184079
4      MrEasy Google Forms Course    Lesson 1               2   FALSE       2   FALSE 1548184085
5      MrEasy Google Forms Course    Lesson 1               2    TRUE       3   FALSE 1548184091
6      MrEasy Google Forms Course    Lesson 1               3      NA       1   FALSE 1548184095
7 Costa Bravo Google Forms Course    Lesson 1               2    TRUE       1   FALSE 1548184151
8 Costa Bravo Google Forms Course    Lesson 1               3      NA       1      NA 1548184154

However, if I pre-fill the responses with something else (I tried a longer text with containing a '?'), then I get errors:
> swirlify::google_form_decode("Lesson1.csv")
Error: 'Z�v��j{f{g���"user","course_name","lesson_name","question_number","correct","attempt","skipped","datetime"
"MrEasy","Google Forms Course","Lesson 1",2,TRUE,1,FALSE,1548183282.43069
"MrEasy","Google Forms Course","Lesson 1",3,NA,1,NA,1548183284.93513
' does not exist in current working directory ('/home/utah113/Teaching/RHUL/Tutorials/R').
In addition: Warning messages:
1: In grepl("\n", x) : input string 1 is invalid in this locale
2: In grepl("\n", path) : input string 1 is invalid in this locale
3: In grepl("^((http|ftp)s?|sftp)://", path) :
  input string 1 is invalid in this locale
4: In grepl("^(/|[A-Za-z]:|\\\\|~)", path) :
  input string 1 is invalid in this locale

So, it would be helpful if the developers explained this part of the process in a bit more detail. Also, when the user completes their session and the log gets passed into the form and the user is prompted to confirm 'Submit' in their browser, the user gets this:
'Your response has been recorded.
Submit another response'
If for whatever reason the user does choose to submit another response, which would be manual, i.e. just typing something in the answer field of the form, then this would normally be uninterpretable for 
swirlify::google_form_decode, resulting in error:
> swirlify::google_form_decode("Untitled form.csv")
Error: '�K$vƣ}�' does not exist in current working directory ('/home/utah113/Teaching/RHUL/Tutorials/R').
In addition: Warning messages:
1: In grepl("\n", x) : input string 1 is invalid in this locale
2: In grepl("\n", path) : input string 1 is invalid in this locale
3: In grepl("^((http|ftp)s?|sftp)://", path) :
  input string 1 is invalid in this locale
4: In grepl("^(/|[A-Za-z]:|\\\\|~)", path) :
  input string 1 is invalid in this locale .
As we cannot do much about this generic behaviour of the google form, perhaps the developers of swirlify could make swirlify::google_form_decode more robust in the face of such unexpected responses/inputs to prevent loss of good responses.  I would appreciate if other users shared their feedback.

**Comment by zackgmani** [comment]
Were you ever able to resolve this issue?

**Comment by Koloydenko** [comment]
Yes, thank you, cannot remember the details though as we stopped using
swirl in 2019.

Kind regards,

Alexey

**Comment by bbitarello** [comment]
Has anyone made progress in solving this?
